### PR TITLE
Encode/Decode `TextColumnProfiler` 

### PR DESCRIPTION
**Pull request by micdavis**
* Added encode/decode for `TextColumnProflier`
* Added Tests

### CODE REVIEW

File: dataprofiler/tests/profilers/test_text_column_profile.py [hunk 1]
```diff
@@ -607,3 +610,135 @@ def test_diff(self):
...
+            [
+                "aa",  # add existing
+                "awsome",  # add new
+            ]
+        )
+
+        # validating update after deserialization
+        deserialized.update(df_str)
+
+        assert deserialized.sample_size == 12
```
Comment (micdavis) [inline dataprofiler/tests/profilers/test_text_column_profile.py:729]: happy to take suggestions on what else I should be asserting here

File: dataprofiler/tests/profilers/test_text_column_profile.py [hunk 2]
```diff
@@ -607,3 +610,135 @@ def test_diff(self):
             places=2,
         )
         self.assertDictEqual(expected_diff, profile_diff)
+
+    @mock.patch("time.time", return_value=0.0)
+    def test_json_encode_after_update(self, time):
+        df = pd.Series(
+            ["abcd", "aa", "abcd", "aa", "b", "4", "3", "2", "dfd", "2"]
```
Comment (micdavis) [inline dataprofiler/tests/profilers/test_text_column_profile.py:617]: wonder if this data is sufficient for testing encode

File: dataprofiler/tests/profilers/test_text_column_profile.py [hunk 3]
```diff
@@ -607,3 +610,135 @@ def test_diff(self):
...
+        deserialized = load_column_profile(json.loads(serialized))
+
+        # Validate reporting after deserialization
+        deserialized.report()
+        test_utils.assert_profiles_equal(deserialized, expected_profile)
+
+        df_str = pd.Series(
+            [
+                "aa",  # add existing
+                "awsome",  # add new
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_text_column_profile.py:722]: this is super clear -- I like

File: dataprofiler/tests/profilers/test_text_column_profile.py [hunk 4]
```diff
@@ -607,3 +610,211 @@ def test_diff(self):
...
+        serialized_dict = json.loads(json.dumps(profiler, cls=ProfileEncoder))
+
+        # popping quantiles and comparing as list below since it is so large
+        serialized_quantiles = serialized_dict["data"].pop("quantiles")
+
+        # popping vocab and comparing as set below since order is random
+        serialized_vocab = serialized_dict["data"].pop("vocab")
+
+        serialized = json.dumps(serialized_dict)
+
```
Comment (micdavis) [inline dataprofiler/tests/profilers/test_text_column_profile.py:648]: add this section so we can test `quantiles` and `vocab`. I use the `profile.quantiles` and `profiler.vocab` as my expected which I think is fair.

File: dataprofiler/tests/profilers/test_text_column_profile.py [hunk 5]
```diff
@@ -607,3 +610,211 @@ def test_diff(self):
...
+    @mock.patch("time.time", return_value=0.0)
+    def test_json_encode_after_update(self, time):
+        self.maxDiff = None
+        df = pd.Series(
+            [
+                "abcd",
+                "aa",
+                "abcd",
+                "lito-potamus",
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_text_column_profile.py:622]: ❤️

File: dataprofiler/tests/profilers/test_text_column_profile.py [hunk 6]
```diff
@@ -607,3 +610,211 @@ def test_diff(self):
             places=2,
         )
         self.assertDictEqual(expected_diff, profile_diff)
+
+    @mock.patch("time.time", return_value=0.0)
+    def test_json_encode_after_update(self, time):
+        self.maxDiff = None
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_text_column_profile.py:616]: remove

**Review by taylorfturner — APPROVED** [review]

**Review by tyfarnan — APPROVED** [review]

**Pull request merged**